### ISSUE TITLE
No way to call set_sorted on LazyFrame in rust

### ISSUE THREAD

**Issue by ewoolsey** [pola-rs/polars#11839]
### Description

Title pretty much says it all. I cannot find any way to call set_sorted for LazyFrame which seems like a fairly important feature. Perhaps I'm missing something?

**Comment by cmdlineluser** [comment]
We can see what the py-polars version does here:

https://github.com/pola-rs/polars/blob/89cc1e2d67448183aa622e1b5f7130793b29b081/py-polars/polars/lazyframe/frame.py#L5648-L5650

It calls `.with_columns(expr.set_sorted())`

https://github.com/pola-rs/polars/blob/89cc1e2d67448183aa622e1b5f7130793b29b081/py-polars/polars/expr/expr.py#L8875

`Expr.set_sorted()` calls the rust function [`.set_sorted_flag()`](https://docs.rs/polars/latest/polars/prelude/enum.Expr.html#method.set_sorted_flag)

So I guess something like `col("*").set_sorted_flag()` (Sorry, don't use the rust version too much.)

**Comment by ewoolsey** [comment]
> We can see what the py-polars version does here:
> 
> https://github.com/pola-rs/polars/blob/89cc1e2d67448183aa622e1b5f7130793b29b081/py-polars/polars/lazyframe/frame.py#L5648-L5650
> 
> It calls `.with_columns(expr.set_sorted())`
> 
> https://github.com/pola-rs/polars/blob/89cc1e2d67448183aa622e1b5f7130793b29b081/py-polars/polars/expr/expr.py#L8875
> 
> `Expr.set_sorted()` calls the rust function [`.set_sorted_flag()`](https://docs.rs/polars/latest/polars/prelude/enum.Expr.html#method.set_sorted_flag)
> 
> So I guess something like `col("*").set_sorted_flag()` (Sorry, don't use the rust version too much.)

Right, but where do you call `col()` from, assuming you have a `LazyFrame`?

**Comment by cmdlineluser** [comment]
Looks like there are some examples in the polars tests:

https://github.com/pola-rs/polars/blob/89cc1e2d67448183aa622e1b5f7130793b29b081/crates/polars/tests/it/lazy/queries.rs#L43-L46

**Comment by ewoolsey** [comment]
> Looks like there are some examples in the polars tests:
> 
> https://github.com/pola-rs/polars/blob/89cc1e2d67448183aa622e1b5f7130793b29b081/crates/polars/tests/it/lazy/queries.rs#L43-L46

Right, but that is adding a new column, I want to set the flag on an existing column. I don't think that's possible?

**Comment by ewoolsey** [comment]
Ahh okay I see that this is possible
```rust
                    .select([
                        col("timestamp").set_sorted_flag(IsSorted::Ascending),
                        col("*").exclude(["timestamp"]),
                    ])
```
Is there a better way to do this? Seems a bit crazy. I would be nice if there was a method that looked like this
```rust
fn(self, exprs: E) -> Self
```
That would just execute the exprs in place and not select anything. Maybe there is and I'm just not aware?

**Comment by cmdlineluser** [comment]
You can use [`.with_column()`](https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html#method.with_column)

> Add a new column to this [DataFrame](https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html) or replace an existing one.

```python
.with_column(col("timestamp").set_sorted_flag(IsSorted::Ascending))
```

**Comment by ewoolsey** [comment]
> You can use [`.with_column()`](https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html#method.with_column)
> 
> > Add a new column to this [DataFrame](https://docs.rs/polars/latest/polars/frame/struct.DataFrame.html) or replace an existing one.
> 
> ```python
> .with_column(col("timestamp").set_sorted_flag(IsSorted::Ascending))
> ```

Omg. The docs on the `LazyFrame::with_column` say this:
```
Add a column, given as an expression, to a DataFrame
```
and mention nothing about replacing... I imagine that's a mistake.

**Comment by stinodego** [comment]
Closed by #11840
